### ISSUE TITLE
When asking for message history and there is none, consider returning an empty [] instead of an error

### ISSUE THREAD

**Issue by mehcode** [Schmavery/facebook-chat-api#146]
Currently we error with:

```
ERR! Error in getThreadHistory { __ar: 1,
ERR! Error in getThreadHistory   error: 1545011,
ERR! Error in getThreadHistory   errorSummary: 'Message Not Found',
ERR! Error in getThreadHistory   errorDescription: 'That message could not be found.',
ERR! Error in getThreadHistory   payload: null,
ERR! Error in getThreadHistory   jsmods: { require: [ [Object] ] },
ERR! Error in getThreadHistory   bootloadable: {},
ERR! Error in getThreadHistory   ixData: {},
ERR! Error in getThreadHistory   lid: '6238580228914591003' }
```

I'd figure it be better to just resolve with `[]`

**Comment by Schmavery** [comment]
Are you searching for a thread that doesn't exist?  As far as I know, https://github.com/Schmavery/facebook-chat-api/blob/master/src/getThreadHistory.js#L36-L38 was supposed to handle the case where you search for a thread with no history, though it's possible facebook has changed since then.

**Comment by bsansouci** [comment]
Agreed, if the thread doesn't exist it should return an error. No message found is a valid response though and should return an empty array.

**Comment by mehcode** [comment]
I'm searching for a thread using a friend ID that doesn't have an matching thread. So, yes, the thread doesn't exist. But the ID is a valid, potential, thread ID. I guess we can't tell the difference though.

**Comment by Schmavery** [comment]
Personally I'm OK with this behaviour.  @bsansouci ?

**Comment by bsansouci** [comment]
Yeah, I think returning an error is appropriate in this case. I'm not a huge fan of the totally wrong error message though, it'd be cool if we could fix those. That said, fixing FB's error messages seems like an whole other can of snails that we don't want to open...

**Comment by mehcode** [comment]
Ok. I can get behind it. As long as it remains easy to identify I can continue ignoring the error.

**Comment by Schmavery** [comment]
I like catching and rethrowing our own error as a good compromise.

**Comment by bsansouci** [comment]
@Schmavery if we re-throw we (finally) have to decide how we're going to identify our own errors so @mehcode can simply ignore this error. Here's a proposal off the top of my head: have a `utils.formatError([string | fberror], id)` that output something of the shape:
```js
{
  errorID: number,
  errorMessage: string,
  isFBError: boolean
}
```
and where `fberror` has the shape:
```js
{ 
  error: number,
  errorSummary: string,
  errorDescription: string,
  ...
}
```

**Comment by Schmavery** [comment]
@bsansouci Lgtm
